spec/models/case_type_spec.rb

```diff
@@ -31,19 +31,19 @@
 
     describe 'is_graduated_fee?' do
       it 'returns false if no fee_type_code' do
-        expect(nil_case_type.is_graduated_fee?).to eql false
+        expect(nil_case_type).not_to be_is_graduated_fee
```

Comment:
(Personal opinion) I am not keen on the expect(subject).to be_flag syntax, it seems convoluted to me, especially when it is not_to be_flag like here. Was the existing code (expect(nil_case_type.is_graduated_fee?).to eql false) raising an offence?

Reply:
This is the sort of thing that made me wonder whether this is the correct policy but yes Rubocop is raising offences over this (more or less).

The original `.to eql false` is picked up by `RSpec/BeEql`, which prefers `be` over `eql`.

I actually changed it to `.to be_falsey`, rather than `.to be false`, and it was then flagged by [`RSpec/PredicateMatcher`](https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/PredicateMatcher), which prefers the `be_is_graduated_fee?`.

As it happens, the `is_graduated_fee?` flag fails the [`Naming/PredicateName`](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/PredicateName) cop but is currently being ignored by the todo file. According to that cop it should be `graduated_fee?` in which case the test would read:

```ruby
expect(nil_case_type).not_to be_graduated_fee
```

and this makes a bit more sense.

Another issue is that I cannot find `is_graduated_fee?` used anywhere other than in the tests. Can it be deleted? @jsugarman

Reply:
This whole section could do with a refactor as it is not using contexts and reusing memoized lets. ideally it would be something like these two separate describes

```
describe '#graduated_fee_type do
   subject  { case_type.graduated_fee_type }
   
   before { create :fixed_fee_type, unique_code: 'FIXED' }
   
   context 'when case type has no fee_type_code' do
      let(:case_type) { build :case_type, fee_type_code: 'FIXED' }
      
      it { is_expected.to be_nil }
   end
...
end
```
and
```
describe '#is_graduated_fee?' do
    subject  { case_type.is_graduated_fee? }
    
    before { create :fixed_fee_type, unique_code: 'FIXED' }
    
    context 'when case type is a fixed fee type' do
      let(:case_type) { build :case_type, fee_type_code: 'FIXED' }
      
      it { is_expected.to be false }
   end
   ...
end
```

Without more investigation into it use or refactoring of the actual method i would be minded to test the actual existing functionality which is to only ever return true or false
```
  def is_graduated_fee?
    graduated_fee_type.nil? ? false : true
  end
```
therefore `it { is_expected.to be false }` and `it { is_expected.to be true }`

>Another issue is that I cannot find is_graduated_fee? used anywhere other than in the tests. Can it be deleted? @jsugarman

I cannot either, nor any potential meta-programming calls to it. I would have a separate PR to remove though

Reply:
BTW, i am not saying you need to refactor but i think `be false` is the best bet